src/Adapter/Product/Image/Repository/ProductImageRepository.php

```diff
@@ -432,4 +444,26 @@ protected function getPreviewCombinationProduct(CombinationId $combinationId): ?
 
         return null;
     }
+
+    public function getAssociatedShopIdsByImageId(ImageId $imageId): array
```

Comment:
```suggestion
    public function getAssociatedShopIds(ImageId $imageId): array
```
Since we are already in the ImageRepository and the only parameter is an `ImageId` I think the last part is a bit redundant It would be different if we had some `getAssociatedShopIds(Image $image)` for example, but we don't so we can keep the name simpler and shorter